FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -1015,4 +1018,15 @@ noncomputable def FiniteAdeleRing.baseChangeContinuousAlgEquiv :
   continuous_invFun := sorry
   -- TODO needs issue number
 
+noncomputable instance baseChangeAlgebra : Algebra K (FiniteAdeleRing B L) :=
+  RingHom.toAlgebra <| (algebraMap L _).comp (algebraMap K L)
```

Comment:
Do you think we should be making all of these Algebra instances? Initially I was doing this but then I switched to semialgebra homs. My motivation was that I wanted to say things like "the natural map from A_K to A_L is more than just a ring homomorphism, it preserves some other structure" and initially I was saying "it's a K-algebra homomorphism" but now I can say "it's a semialgebra morphism over the natural map K -> L" and this avoids having to make A_L a K-algebra.

Reply:
Oh that's funny! So we need that L tensor_K A_K -> A_L is L-linear, K-linear and A_K-linear! No that's fine, you've convinced me that this is useful.